systems/plants/RigidBodyManipulator.cpp

```diff
@@ -1351,89 +1366,17 @@ void RigidBodyManipulator::doKinematicsNew(const MatrixBase<DerivedQ>& q, const
   jdotV_cached = compute_JdotV && velocity_kinematics_cached;
-  for (int i = 0; i < num_positions; i++) cached_q[i] = q[i];
-  if (v.rows() > 0) for (int i = 0; i < num_velocities; i++) cached_v[i] = v[i];
-}
-
-template <typename DerivedA, typename DerivedB>
-void RigidBodyManipulator::getCMM(MatrixBase<DerivedA> const & q, MatrixBase<DerivedA> const & qd, MatrixBase<DerivedB> &A, MatrixBase<DerivedB> &Adot)
-{
-  if (use_new_kinsol) {
-    warnOnce("new_kinsol_old_method_getCMM", "Warning: called old getCMM with use_new_kinsol set to true.");
-    typedef typename DerivedB::Scalar Scalar;
-    GradientVar<Scalar, TWIST_SIZE, Eigen::Dynamic> cmm = centroidalMomentumMatrix<Scalar>(1);
-    A = cmm.value();
-    VectorXd Adot_vectorized = cmm.gradient().value() * qd;
-    Map<typename MatrixBase<DerivedA>::PlainObject> Adot_map(Adot_vectorized.data(), Adot.rows(), Adot.cols());
-    Adot = Adot_map;
-    return;
-  }
-
-  // returns the centroidal momentum matrix as described in Orin & Goswami 2008
-  //
-  // h = A*qd, where h(4:6) is the total linear momentum and h(1:3) is the
-  // total angular momentum in the centroid frame (world fram translated to COM).
-
-  Vector3d com; getCOM(com);
-  Xtrans(-com,&Xcom);
-
-  getCOMJac(Jcom);
-  Vector3d com_dot = Jcom*qd;
-  dXcom = MatrixXd::Zero(6,6);
-  dXcom(5,1) = 1*com_dot(0);
-  dXcom(4,2) = -1*com_dot(0);
-  dXcom(5,0) = -1*com_dot(1);
-  dXcom(3,2) = 1*com_dot(1);
-  dXcom(4,0) = 1*com_dot(2);
-  dXcom(3,1) = -1*com_dot(2);
-
-  A = MatrixXd::Zero(6,num_positions);
-  Adot = MatrixXd::Zero(6,num_positions);
-
-  for (int i=0; i < NB; i++) {
-    Ic[i] = I[i];
-    dIc[i] = MatrixXd::Zero(6,6);
-  }
-
-  int n;
-  for (int i=NB-1; i >= 0; i--) {
-    n = dofnum[i];
-    jcalc(pitch[i],q[n],&Xi,&phi[i]);
-    Xup[i] = Xi * Xtree[i];
 
-    djcalc(pitch[i], q[n], &dXidq);
-    dXup[i] = dXidq * Xtree[i] * qd[n];
+  cached_q = q;
+  cached_q_new = q;
 
-    if (parent[i] >= 0) {
-      Ic[parent[i]] += Xup[i].transpose()*Ic[i]*Xup[i];
-      dIc[parent[i]] += (dXup[i].transpose()*Ic[i] + Xup[i].transpose()*dIc[i])*Xup[i] + Xup[i].transpose()*Ic[i]*dXup[i];
-    }
-  }
-
-
-  for (int i=0; i < NB; i++) {
-    if (parent[i] >= 0) {
-      Xworld[i] = Xup[i] * Xworld[parent[i]];
-      dXworld[i] = dXup[i]*Xworld[parent[i]] + Xup[i]*dXworld[parent[i]];
-    }
-    else {
-      Xworld[i] = Xup[i];
-      dXworld[i] = dXup[i];
-    }
-
-    Xg = Xworld[i] * Xcom; // spatial transform from centroid to body
-    dXg = dXworld[i] * Xcom + Xworld[i] * dXcom;
-
-    n = dofnum[i];
-    A.col(n) = Xg.transpose()*Ic[i]*phi[i];
-    Adot.col(n) = dXg.transpose()*Ic[i]*phi[i] + Xg.transpose()*dIc[i]*phi[i];
+  if (v.rows() > 0) {
+    cached_v = v;
+    cached_v_new = v;
   }
 }
 
 void RigidBodyManipulator::updateCompositeRigidBodyInertias(int gradient_order) {
-  if (!use_new_kinsol) {
-    throw std::runtime_error("method requires new kinsol format");
-  }
```

Comment:
Got rid of these. This is now be handled in checkCachedKinematicsSettings by virtue of the fact that position_kinematics_cached would be false otherwise